### PR TITLE
feat: Add date-range filter on CSAT Reports

### PR DESCRIPTION
**Pull request by pranavrajs**
- Move date range to a separate component
- Add filter on the store 
- Display dates on the table

Screenshot 2021-07-14 at 3 21 21 PM

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/settings/reports/components/DateRangeSelector.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<template>
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/reports/components/DateRangeSelector.vue:1]: Good candidate for storybook.

**Review by nithindavid — APPROVED** [review]
👍

**Comment by pranavrajs** [comment]
@muhsin-k Can you please test both reports and merge it?

**Comment by muhsin-k** [comment]
@pranavrajs  The functionality is working as expected. Can we show the items by created time in descending order? 
Screenshot 2021-07-14 at 7 28 48 PM

**Comment by pranavrajs** [comment]
@muhsin-k Thanks for pointing it out. I have made the changes, please take a look.

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**